blockchain/fullblocktests/generate.go

```diff
@@ -2133,3 +2147,142 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 
 	return tests, nil
 }
+
+type TestHeaderInstance interface {
```

Comment:
There's no comments on any of these exported structs and functions

Reply:
Added comments and moved the header structs along with the block structs, so all of them can be referred. And added comments for the functions inside GenerateHeaders